fe/fe-core/src/main/java/com/starrocks/alter/Alter.java

```diff
@@ -436,6 +440,15 @@ public void processAlterTable(AlterTableStmt stmt) throws UserException {
                 } else {
                     throw new DdlException("Invalid alter opertion: " + alterClause.getOpType());
                 }
+            } else if (currentAlterOps.hasTruncatePartitionOp()) {
+                // this logic is use to adapt mysql syntax, and cannot be executed after adding a write lock
+                // ALTER TABLE test TRUNCATE PARTITION p1;
+                db.writeUnlock();
+                TruncatePartitionClause clause = (TruncatePartitionClause) alterClauses.get(0);
+                TableRef tableRef = new TableRef(stmt.getTbl(), null, clause.getPartitionNames());
+                TruncateTableStmt tStmt = new TruncateTableStmt(tableRef);
```

Comment:
It's better to change the truncateTable. Because truncatePartition does not need the DB lock.